storage/src/main/java/com/firebase/ui/storage/images/FirebaseImageLoader.java

```diff
@@ -0,0 +1,82 @@
+package com.firebase.ui.storage.images;
+
+import android.util.Log;
+
+import com.bumptech.glide.Priority;
+import com.bumptech.glide.load.data.DataFetcher;
+import com.bumptech.glide.load.model.stream.StreamModelLoader;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.storage.StorageReference;
+import com.google.firebase.storage.StreamDownloadTask;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * ModelLoader implementation to download images from FirebaseStorage with Glide.
+ *
+ * Sample Usage:
+ * <pre>
+ *     StorageReference ref = FirebaseStorage.getInstance().getReference().child("myimage");
+ *     ImageView iv = (ImageView) findViewById(R.id.my_image_view);
+ *
+ *     Glide.with(this)
+ *         .using(new FirebaseImageLoader())
+ *         .load(ref)
+ *         .into(iv);
+ * </pre>
+ */
+public class FirebaseImageLoader implements StreamModelLoader<StorageReference> {
+
+    private static final String TAG = "FirebaseImageLoader";
+
+    @Override
+    public DataFetcher<InputStream> getResourceFetcher(StorageReference model, int width, int height) {
+        return new FirebaseStorageFetcher(model);
+    }
+
+    private class FirebaseStorageFetcher implements DataFetcher<InputStream> {
+
+        private StorageReference mRef;
+        private StreamDownloadTask mStreamTask;
+        private InputStream mInputStream;
+
+        FirebaseStorageFetcher(StorageReference ref) {
+            mRef = ref;
+        }
+
+        @Override
+        public InputStream loadData(Priority priority) throws Exception {
+            mStreamTask = mRef.getStream();
+            mInputStream = Tasks.await(mStreamTask).getStream();
```

Comment:
When I call `.await()` that will throw an `Exception` if something goes wrong.  That will be handled by Glide, and the user can attach failure listeners to Glide if they want to know about this.